### PR TITLE
28668 expression editor take two

### LINKED ISSUE #28668: [Epic] De-crazy the expression editor function helper popover
We want to improve expression editor helper popover to make it less overwhelming and more clean and helpful.
To achieve that we want to change it's design and structure.

<table>
<tr>
<th> Before: </th>
<th> After: </th>
</tr>
<tr>
<td>

Screenshot 2023-02-27 at 19 34 09

</td>
<td>

Screenshot 2023-02-27 at 19 41 14

</td>
</tr>
</table>

**Links**
- [product doc](https://www.notion.so/metabase/De-crazy-the-expression-editor-function-helper-popover-c43d8ff0b0ca4271b9c36ec2be844fe0)
- [design](https://www.figma.com/file/68g8S9L4cMesbctGfCySl7/De-crazy-the-expression-editor-function-helper-popover?node-id=0%3A1&t=1d5CQFMXJ6PhbMu8-0)
- feature branch: `28668-expression-editor`

**Implementation Plan**

- [x] [Refactored ExpressionWidget, ClauseStep components to TS](https://github.com/metabase/metabase/pull/28687)
- [x] [Change expression editor title, Implement info tooltip](https://github.com/metabase/metabase/pull/28776) 
- [x] [Change popover structure to match proposed design](https://github.com/metabase/metabase/pull/28920)
- [x] [Add interactivity for popover parameters](https://github.com/metabase/metabase/pull/28966)

**How to Test This Feature Out**

1. Start creating a new Question: click "New" button in the top bar -> Question.
2. Click "Custom column" in the Notebook.
3. Appeared expression editor should have new design and experience.

### PR DESCRIPTION
**Pull request by deniskaber**
Closes https://github.com/metabase/metabase/issues/28668 second iteration, check it for full details

### Description

We want to improve expression editor helper popover to make it less overwhelming and more clean and helpful.
To achieve that we want to change it's design and structure.

<table>
<tr>
<th> Before: </th>
<th> After: </th>
</tr>
<tr>
<td>

Screenshot 2023-02-27 at 19 34 09

</td>
<td>

Screenshot 2023-02-27 at 19 34 09

</td>
</tr>
</table>

### How to verify

1. Start creating a new Question: click "New" button in the top bar -> Question.
2. Click "Custom column" in the Notebook.
3. Appeared expression editor should have new design and experience.

### Demo

https://user-images.githubusercontent.com/7983744/230657659-fd925de1-0098-4689-be05-7a7df1799c16.mov

### Checklist

- [x] Add new design - restore link to documentation, add details for every argument

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29893)

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.styled.tsx [hunk 1]
```diff
@@ -1,38 +1,71 @@
...
-  font-weight: 700;
+  display: grid;
+  gap: 1rem;
+  grid-template-columns: repeat(2, minmax(2rem, max-content));
+`;
+
+export const ArgumentItem = styled.div`
+  color: ${color("accent3")};
+
+  font-size: 12px;
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.styled.tsx:40]: Please change `font-size` and `line-height` to rem

**Review by ranquild — APPROVED** [review]
@deniskaber Looks good from the FE point of view, please double check with @mazameli if everything works as expected

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.styled.tsx (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.styled.tsx:40]: is it a requirement? @ranquild

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.unit.spec.tsx [hunk 2]
```diff
@@ -105,5 +85,10 @@ function setup(additionalProps?: Partial<ExpressionEditorHelpTextProps>) {
 
   render(<ExpressionEditorHelpText {...props} />);
 
+  // have to wait for TippyPopover to render content
+  await waitFor(() => {
+    expect(screen.getByTestId("expression-helper-popover")).toBeInTheDocument();
```
Comment (uladzimirdev) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.unit.spec.tsx:90]: I'm not sure is it a case here, but my understanding is to use `waitFor` when I need to be sure function is called (e.g. toHaveBeenCalled), but `expect(await screen.findbBy...).toBeInTheDocument()` when we work with DOM

do you know why  `await waitFor` can be better than `findBy`?

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.styled.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/query_builder/components/expressions/ExpressionEditorHelpText.styled.tsx:40]: I believe we almost always use `rem` for font and space

**Review by mazameli — CHANGES_REQUESTED** [review]
Looking directionally good, but some font details got missed: 
image

**Review by mazameli — APPROVED** [review]
Looking good!

**Pull request merged**